### PR TITLE
Adding karma package

### PR DESCRIPTION
**Pull request by utieyin**
* "karma-0.120.0: new package"

Fixes:

Related:

#### For new package PRs only

- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates

### CODE REVIEW

File: karma.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# melange.yaml
+package:
+  name: karma
+  version: "0.102"
```
[the remark this answers is not in the document]
Comment (utieyin) [reply karma.yaml]: fixed

File: karma.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# melange.yaml
```
Comment (jamonation) [inline karma.yaml:1]:
```suggestion
```

File: karma.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  epoch: 0
+  copyright:
+    - license: Apache-2.0
+
+environment:
+  contents:
+    packages:
+      - busybox
+      - ca-certificates-bundle
+      - go
```
[the remark this answers is not in the document]
Comment (utieyin) [reply karma.yaml]: fixed

**Review by jamonation — DISMISSED** [review]

File: karma.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    - name: Run Karma test
+      runs: |
+        karma --version
+    - name: "Check karma status"
+      uses: test/daemon-check-output
+      with:
+        start: "${{package.name}}"
+        timeout: 60
+        expected_output: |
+          Starting HTTP server
```
Comment (Dentrax) [inline karma.yaml:57]: If there is endpoint, can we do a simple `curl | grep` test here in the `post:`?

File: karma.yaml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      expected-commit: 8009f4aa581980654e5c67b26b41e85d9e7ccce0
+
+  - name: install-npm-dependencies
+    working-directory: ui
+    pipeline:
+      - runs: |
+          mkdir -p node_modules/.bin
+          touch node_modules/.install
+          npm install --package-lock-only
+          npm ci
```
Comment (Dentrax) [inline karma.yaml:31]:
```suggestion
          npm ci --omit=dev
```

File: karma.yaml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+          cat node_modules/bootstrap/scss/_root.scss | sed s/':root {'/'* {'/ > src/Styles/BootstrapRoot.scss
```
Comment (Dentrax) [inline karma.yaml:32]: Can we also add a new line above, (just after `npm ci`)
```
npm prune --omit=dev
```

To prune unnecessary things?

**Review by Dentrax — DISMISSED** [review]

File: karma.yaml (hunk 4 above)
Comment (utieyin) [reply karma.yaml:62]: Done. Added a check to test `/health` endpoint

File: karma.yaml (hunk 5 above)
Comment (utieyin) [reply karma.yaml:32]: change implemented

Comment (utieyin) [reply karma.yaml:32]: The build fails when `npm ci` is run for only production either by adding `--omit=dev` or `--production`

Comment (Dentrax) [reply karma.yaml:32]: Its weird, what was the issue? We're just omitting the dev packages that don't need to be exist on the runtime.

**Comment by jamonation** [comment]
@Dentrax I suspect it is because some of the build dependencies show up in `devDependencies` e.g. https://github.com/prymitive/karma/blob/293e431c30e9e50e11809236377e756262326012/ui/package.json#L89-L90

**Review by jamonation — APPROVED** [review]

**Review by Dentrax — APPROVED** [review]

**Pull request merged**